### PR TITLE
Management.Sql - Generated SQL DataSync and added test scenario's

### PR DESCRIPTION
**Pull request by ScottHolden**
## Description
Generated Management.Sql with package-2015-05-preview api spec including datasync, added SyncAgent SyncGroup SyncMember scenario's with session records.

Corresponding rest-api-spec change can be found at:
https://github.com/Azure/azure-rest-api-specs/pull/1510

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by ScottHolden** [comment]
I have signed the cla :)

**Comment by shahabhijeet** [comment]
@ScottHolden are you part of the Azure org, if not please become a part of the Azure Team by visiting
https://github.com/Azure/

**Comment by jaredmoo** [comment]
@fhljys ,@ferno-ms , @nathannfan fyi

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <PackageId>Microsoft.Azure.Management.Sql</PackageId>
     <Description>Azure SQL Management SDK library</Description>
     <AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-    <VersionPrefix>1.7.0-preview</VersionPrefix>
+    <VersionPrefix>1.8.0-preview</VersionPrefix>
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: the latest published version is 1.6.0 (see https://www.nuget.org/packages/Microsoft.Azure.Management.Sql ) so please undo this

File: src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -22,8 +22,4 @@
 [assembly: AssemblyTitle("Microsoft Azure SQL Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure SQL.")]
 [assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.7.0.0")]
-
-
-
-
+[assembly: AssemblyFileVersion("1.8.0.0")]
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs]: please undo

File: src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                SyncMember updateSyncMember = sqlClient.SyncMembers.Update(resourceGroup.Name, server.Name, testDatabaseName, syncGroupName, syncMemberName, new SyncMember
+                {
+                    SyncDirection = updateSyncMemberDirection,
+                    DatabaseType = syncMemberDatabaseType,
+                    UserName = SqlManagementTestUtilities.DefaultLogin,
+                    Password = SqlManagementTestUtilities.DefaultPassword,
+                });
+                Assert.NotNull(updateSyncMember);
+                Assert.Equal(updateSyncMemberDirection, updateSyncMember.SyncDirection);
+                Assert.NotEqual(syncMemberDirection, updateSyncMemberDirection);
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs]: can you also test empty update, e.g.

```
// Empty update of sync member
SyncMember updateSyncMember2 = sqlClient.SyncMembers.Update(resourceGroup.Name, server.Name, testDatabaseName, syncGroupName, syncMemberName, new SyncMember());
```

File: src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                Assert.Equal(syncMemberDirection, getSyncMember.SyncDirection);
+                Assert.Equal(syncMemberDatabaseType, getSyncMember.DatabaseType);
+                Assert.Equal(memberDatabaseName, getSyncMember.DatabaseName);
+                Assert.Equal(server.Name, getSyncMember.ServerName);
+
+                // List sync members
+                IPage<SyncMember> listSyncMembers = sqlClient.SyncMembers.ListBySyncGroup(resourceGroup.Name, server.Name, testDatabaseName, syncGroupName);
+                Assert.NotNull(listSyncMembers);
+                Assert.Equal(1, listSyncMembers.Count());
+                Assert.Equal(syncMemberName, listSyncMembers.FirstOrDefault()?.Name);
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs]: nit: already checked that there is one, so this can be `listSyncMembers.Single().Name`

File: src/SDKs/SqlManagement/Sql.Tests/SyncGroupScenarioTests.cs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                Assert.Equal(syncGroupName, listSyncGroups.FirstOrDefault()?.Name);
+
+                // Update sync group
+                int interval2 = 600;
+                SyncGroup updateSyncGroup = sqlClient.SyncGroups.Update(resourceGroup.Name, server.Name, testDatabaseName, syncGroupName, new SyncGroup
+                {
+                    Interval = interval2
+                });
+                Assert.NotNull(updateSyncGroup);
+                Assert.Equal(interval2, updateSyncGroup.Interval);
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/SyncGroupScenarioTests.cs]: Can you please test empty update, i..e.

```
SyncGroup updateSyncGroup = sqlClient.SyncGroups.Update(resourceGroup.Name, server.Name, testDatabaseName, syncGroupName, new SyncGroup());
```

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Looks excellent, just some minor changes requested. Thank you!

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj (hunk 1 above)
Comment (ScottHolden) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Done :)

File: src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (ScottHolden) [inline src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs]: Done :)

File: src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs (hunk 4 above)
Comment (ScottHolden) [inline src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs]: Done :)

File: src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs (hunk 3 above)
Comment (ScottHolden) [inline src/SDKs/SqlManagement/Sql.Tests/SyncMemberScenarioTests.cs]: Done :)

File: src/SDKs/SqlManagement/Sql.Tests/SyncGroupScenarioTests.cs (hunk 5 above)
Comment (ScottHolden) [inline src/SDKs/SqlManagement/Sql.Tests/SyncGroupScenarioTests.cs]: Done :)

**Comment by ScottHolden** [comment]
@shahabhijeet I'm not currently part of the Azure GitHub organisation, who do I need to reach out to to be added, as I couldn't see a link on https://github.com/Azure/

**Comment by jaredmoo** [comment]
@ScottHolden I think that this is only applicable to Microsoft employees working in the Azure organization. So my understanding is that this is not applicable to you.

File: src/SDKs/SqlManagement/Sql.Tests/SyncGroupScenarioTests.cs (hunk 5 above)
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/SyncGroupScenarioTests.cs]: oops, these empty updates ideally should not fail. Not your fault though, it has to be fixed server-side. Thanks for testing!

**Review by jaredmoo — APPROVED** [review]

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
@ScottHolden add to whitelist

**Comment by shahabhijeet** [comment]
@ScottHolden sql tests are failing on travis.
Were you able to successfully run SQL tests locally?
Click on Details and check the failures.

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by ScottHolden** [comment]
All tests were working fine locally @shahabhijeet. I was able to run in record mode and playback mode, and committed the SessionRecord with the test changes.
I'll have a look into what is happening and try and replicate locally :)

**Comment by ScottHolden** [comment]
Running locally gives me no errors at all, I can't seem to replicate :(

image

**Comment by jaredmoo** [comment]
It looks like the lab build merges with the target branch (azure/psSdkJson6) before running tests, and we made some incompatible test changes recently that you haven't pulled in. It should be fixed by merging from azure/psSdkJson6 and re-recording the tests. You will need to add a few new properties to your test environment connection string, e.g. `DefaultLocationId=japaneast;DefaultLocation=Japan East`.

**Comment by ScottHolden** [comment]
Thanks @jaredmoo ! That did the trick :)
Should be fine to merge now :)

**Review by shahabhijeet — APPROVED** [review]
@jaredmoo I assume you will be updating the AssemblyInfo to reflect the next release version.

**Comment by jaredmoo** [comment]
Awesome! Yes AssemblyInfo is handled. Thx
